c10/core/impl/InlineEvent.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "c10/core/impl/EventInterface.h"
+#include "c10/core/DeviceType.h"
+#include "c10/core/Stream.h"
+#include "c10/util/Exception.h"
+
+namespace c10 {
+namespace impl {
+
+template <typename T>
+struct InlineEvent final : public EventInterface {
```

Comment:
So, when do we get `struct CUDAEvent : InlineEvent<CUDAGuardImpl>` ;)

Reply:
I didn't want to collide with ATen's CUDAEvent, which contains additional backend-specific functionality.